packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -194,6 +203,54 @@ class UpdatePackagesCommand extends FlutterCommand {
       );
     }
 
+    if (forceUpgrade && cherryPickPackage != null) {
+      throwToolExit(
+          '--force-upgrade cannot be used with the --cherry-pick-package flag'
+      );
+    }
+
+    if (forceUpgrade && isPrintPaths) {
+      throwToolExit(
+          '--force-upgrade cannot be used with the --paths flag'
+      );
+    }
+
+    if (forceUpgrade && isPrintTransitiveClosure) {
+      throwToolExit(
+          '--force-upgrade cannot be used with the --transitive-closure flag'
+      );
+    }
+
+    if (cherryPickPackage != null && offline) {
+      throwToolExit(
+          '--cherry-pick-package cannot be used with the --offline flag'
+      );
+    }
+
+    if (cherryPickPackage == null && cherryPickVersion != null) {
```

Comment:
what if `cherryPickPackage != null && cherryPickVersion == null`?

Reply:
Ahh ok, I see how this is supported, see my later comment https://github.com/flutter/flutter/pull/128917/files#r1230327031